thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/detector/email/filter/AlertFilterFactory.java

```diff
@@ -42,26 +41,47 @@ private void loadPropertiesFromInputStream(InputStream input) {
     try {
       props.load(input);
     } catch (IOException e) {
-      LOGGER.error("Error loading the functions from config", e);
+      LOGGER.error("Error loading the alert filters from config", e);
     } finally {
       IOUtils.closeQuietly(input);
     }
 
-    LOGGER.info("Found {} entries in alert filter autotune configuration file {}", props.size());
+    LOGGER.info("Found {} entries in alert filter configuration file {}", props.size());
     for (Map.Entry<Object, Object> entry : props.entrySet()) {
       LOGGER.info("{}: {}", entry.getKey(), entry.getValue());
     }
   }
 
-  public AlertFilter fromSpec(AnomalyFunctionDTO anomalyFunctionSpec) throws Exception {
-    AlertFilter alertFilter = null;
-    String type = anomalyFunctionSpec.getType();
-    if (!props.containsKey(type)) {
-      throw new IllegalArgumentException("Unsupported type " + type);
+
+  /**
+   * Initiates an alert filter for the given anomaly function.
+   *
+   * @param anomalyFunctionSpec the anomaly function that contains the alert filter spec.
+   *
+   * @return the alert filter specified by the alert filter spec or a dummy filter if the function
+   * does not have an alert filter spec or this method fails to initiates an alert filter from the
+   * spec.
+   */
+  public AlertFilter getAlertFilter(AnomalyFunctionBean anomalyFunctionSpec) {
+    Map<String, String> alertFilterInfo = anomalyFunctionSpec.getAlertFilter();
+    if (alertFilterInfo == null) {
+      alertFilterInfo = Collections.emptyMap();
+    }
+    AlertFilter alertFilter = new DummyAlertFilter();
+    if (alertFilterInfo.containsKey(FILTER_TYPE_KEY)) {
+      String alertFilterType = alertFilterInfo.get(FILTER_TYPE_KEY);
+      if(props.containsKey(alertFilterType.toUpperCase())) {
+        String className = props.getProperty(alertFilterType.toUpperCase());
+        try {
+          alertFilter = (AlertFilter) Class.forName(className).newInstance();
+        } catch (Exception e) {
+          LOGGER.info(e.getMessage());
+        }
+      } else if(alertFilterType.equals(ALPHA_BETA_ALERT_FILTER)) {
```

Comment:
This "else if" block should also be taken care of in previous if block. The props should contain the mapping from all filter names to their class path regardless the filter is implemented in TE or anomaly library.